### ISSUE TITLE
bug: Not showing contents of "* nor "+ in OS clipboard on Ubuntu 24.04LTS

### ISSUE THREAD

**Issue by chupe** [vscode-neovim/vscode-neovim#2261]
### Check the following:

- [X] I have tried setting `vscode-neovim.neovimClean` in VSCode settings and **restarting VSCode.**
- [X] I have read the vscode-neovim docs.
- [X] I have searched existing (closed!) issues.

### Neovim version (nvim -v)

0.9.5 AND 0.10.1

### Operating system/version

Ubuntu 24.04LTS

### Problem (including [log messages](https://github.com/vscode-neovim/vscode-neovim?tab=readme-ov-file#troubleshooting))

No content can be copied to OS clipboard. `y`, `c`, `d` etc. don't move content to the system clipboard.

I tried running with `--clean`, tried updating to latest versions of `nvim` (0.9.5 > 0.10.1), `code` (1.93 > 1.93.1) and the extension (1.18.09 > 1.18.12).
Also tried disabling all extensions and running on clean vscode installation (`set clipboard=unnamedplus` manually after the fact).

Specifying the register (`"*yy` or `"+yy`) doesn't help either.

`:reg &clipboard` returns `unnamedplus`
`:reg` returns:
```
Type Name Content
  l  ""   // nvim clipboard test^J
  l  "0   // nvim clipboard test^J

  l  "*   // nvim clipboard test^J

  l  "+   // nvim clipboard test^J
  c  ".   ^@// nvim clipboard test
  c  ":   re
```

All of the expected behavior works when running `nvim` in the terminal.

Log file while reproducing:
```
2024-09-24 10:17:11.259 [info] MainController: Starting nvim: /opt/nvim-linux64/bin/nvim -N --embed --cmd source /home/x/.config/Code/extensions/asvetliakov.vscode-neovim-1.18.12/runtime/vscode-neovim.vim --clean
2024-09-24 10:17:11.294 [info] MainController: Nvim info: {
  configDir: '/home/x/.config/nvim',
  nvimVersion: {
    major: 0,
    minor: 10,
    patch: 1,
    prerelease: false,
    build: 'g7e194f0d0',
    api_prerelease: false,
    api_compatible: 0,
    api_level: 12
  },
  logFile: '/home/x/.local/state/nvim/log'
}
2024-09-24 10:17:50.818 [info] MessagesManager: '\n' +
  'Type Name Content\n' +
  '  l  ""   // nvim clipboard test^J\n' +
  '  l  "0   // nvim clipboard test^J\n'
2024-09-24 10:17:50.820 [info] MessagesManager: '\n' +
  'Type Name Content\n' +
  '  l  ""   // nvim clipboard test^J\n' +
  '  l  "0   // nvim clipboard test^J\n' +
  '\n' +
  '  c  "*   NVIM v0.9.5^JNVIM v0.10.1\n'
2024-09-24 10:17:50.820 [info] MessagesManager: '\n' +
  'Type Name Content\n' +
  '  l  ""   // nvim clipboard test^J\n' +
  '  l  "0   // nvim clipboard test^J\n' +
  '\n' +
  '  c  "*   NVIM v0.9.5^JNVIM v0.10.1\n' +
  '\n' +
  '  c  "+   NVIM v0.9.5^JNVIM v0.10.1\n' +
  '  c  "%   vscode-userdata:/home/x/.config/Code/data/User/settings.json\n'
2024-09-24 10:18:27.431 [info] MessagesManager: '\nType Name Content\n  c  ""   clipboard\n  c  "0   clipboard\n'
2024-09-24 10:18:27.432 [info] MessagesManager: '\n' +
  'Type Name Content\n' +
  '  c  ""   clipboard\n' +
  '  c  "0   clipboard\n' +
  '\n' +
  '  c  "*   clipboard\n'
2024-09-24 10:18:27.432 [info] MessagesManager: '\n' +
  'Type Name Content\n' +
  '  c  ""   clipboard\n' +
  '  c  "0   clipboard\n' +
  '\n' +
  '  c  "*   clipboard\n' +
  '\n' +
  '  c  "+   clipboard\n' +
  '  c  ":   set clipboard=unnamedplus\n' +
  '  c  "%   vscode-userdata:/home/x/.config/Code/data/User/settings.json\n'
```

### Steps To Reproduce

On Ubuntu 24.04 LTS try using OS clipboard with any motion that saves to unnamed register.

### Expected Behavior

Expect to see the contents from the editor inside the OS clipboard like when using `nvim` in the terminal.

**I also want to thank you for your time. Really appreciate the work you guys are putting into this.
It helps bridge the gap and get best of the both worlds.**

**Comment by justinmk** [comment]
Excellent report, thanks! This likely comes down to $PATH or some other problem preventing Nvim from finding the "clipboard tool". From vscode, trying running :checkhealth to see Nvim's report and see what it says in the Clipboard section (note: vscode-neovim's handling of the checkhealth buffer seems buggy, so / or ctrl-f might not work even though the Clipboard section is there).

Also try `:echom $PATH` from vscode-neovim and compare that to `nvim` in the terminal.

Other than that, `:help clipboard` gives ways to set the clipboard tool explicitly.

**Comment by chupe** [comment]
Turns out I was missing all of the clipboard utilities listed under `:help clipboard`.
Something I might have done after upgrading from Ubuntu 22 to 24 since some issues with the clipboard started popping up - so I uninstalled anything clipboard-related until the issues went away, except for the clipboard manager that I use.

The issue of the missing clipboard utility was disguised from the terminal since I am usually running `tmux` which stepped in as a clipboard utility.

Thank you @justinmk for the help!
